### ISSUE TITLE
Implement RenderSimpleIdentifier/RenderSimpleParams [phab:D1668]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2092]
_Created by **schroederc** at 2017-05-23 20:20:33_: 

RenderSimpleIdentifier and RenderSimpleParams should be equivalent to
the C++ implementations and are exposed in integration tests by the
/kythe/code/rendered/identifier and /kythe/code/rendered/params facts
for nodes with a MarkedSource.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-05-23 20:31:08_:

Please document what this does (it's kind of clear what it means, but not what effect that has if the result is deeper than that).

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-05-23 20:31:08_:

Maybe make the buffer part of this type, so that `render` and `renderIdent` can get them from the state it needs anyway?

```
type state struct {
   w io.Writer
   depth Int
   inIdent bool
}

func (st *state) render(ms *cpb.MarkedSource) {…
```

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-05-23 20:31:08_:

Please document how this test works (i.e., how it verifies the behaviour we expect).

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-05-23 20:54:04_:

Doc

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-05-23 20:55:59_:

Field

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-05-23 20:58:59_:

More doc

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2017-05-23 20:59:14_:

This was copied from the C++ version as an attempt to be completely inoperable.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2017-05-23 21:06:13_:

s/inoperable/interoperable/

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2017-05-23 22:50:00_

**Comment by kythe** [comment]
_**schroederc** closed via commit f3fa37e1de627b90636cf42f05734f402eb54ddb at 2017-05-23 22:50:55_
